frontend/src/metabase/common/components/DatePicker/RelativeDatePicker/DateIntervalPicker/DateIntervalPicker.tsx

```diff
@@ -97,13 +106,15 @@ export function DateIntervalPicker({
               <Menu.Item
                 icon={<Icon name="arrow_left_to_line" />}
                 onClick={handleStartingFromClick}
+                onMouseDown={handleStartingFromClick}
               >
                 {t`Starting from…`}
               </Menu.Item>
             )}
             <Menu.Item
               icon={<Icon name={includeCurrent ? "check" : "calendar"} />}
               onClick={handleIncludeCurrentClick}
+              onMouseDown={handleIncludeCurrentClick} // we need to close the menu on mousedown to prevent the outer popover from closing
```

Comment:
I don't think this is an acceptable solution - the UX is weird.

This fixes it nicely:
```diff
diff --git a/frontend/src/metabase/ui/components/overlays/Menu/Menu.styled.tsx b/frontend/src/metabase/ui/components/overlays/Menu/Menu.styled.tsx
index 2e3693cf2c..853a44cc83 100644
--- a/frontend/src/metabase/ui/components/overlays/Menu/Menu.styled.tsx
+++ b/frontend/src/metabase/ui/components/overlays/Menu/Menu.styled.tsx
@@ -6,10 +6,11 @@ export const getMenuOverrides = (): MantineThemeOverride["components"] => ({
     defaultProps: {
       radius: "sm",
       shadow: "md",
-      withinPortal: true,
+      withinPortal: false,
     },
     styles: theme => ({
       dropdown: {
+        position: "fixed",
         padding: "0.75rem !important",
         minWidth: "11.5rem",
       },
```

plus, of course:
```diff
diff --git a/frontend/src/metabase/common/components/DatePicker/RelativeDatePicker/DateIntervalPicker/DateIntervalPicker.tsx b/frontend/src/metabase/common/components/DatePicker/RelativeDatePicker/DateIntervalPicker/DateIntervalPicker.tsx
index 8e687e9981..7701e00e36 100644
--- a/frontend/src/metabase/common/components/DatePicker/RelativeDatePicker/DateIntervalPicker/DateIntervalPicker.tsx
+++ b/frontend/src/metabase/common/components/DatePicker/RelativeDatePicker/DateIntervalPicker/DateIntervalPicker.tsx
@@ -42,7 +42,6 @@ export function DateIntervalPicker({
   onChange,
   onSubmit,
 }: DateIntervalPickerProps) {
-  const [isMenuOpened, setIsMenuOpened] = useState(false);
   const interval = getInterval(value);
   const unitOptions = getUnitOptions(value);
   const includeCurrent = getIncludeCurrent(value);
@@ -63,12 +62,10 @@ export function DateIntervalPicker({
 
   const handleStartingFromClick = () => {
     onChange(setDefaultOffset(value));
-    setIsMenuOpened(false);
   };
 
   const handleIncludeCurrentClick = () => {
     onChange(setIncludeCurrent(value, !includeCurrent));
-    setIsMenuOpened(false);
   };
 
   const handleSubmit = (event: FormEvent) => {
@@ -92,7 +89,7 @@ export function DateIntervalPicker({
           ml="md"
           onChange={handleUnitChange}
         />
-        <Menu opened={isMenuOpened} onChange={setIsMenuOpened}>
+        <Menu>
           <Menu.Target>
             <Button
               c="text.2"
@@ -106,7 +103,6 @@ export function DateIntervalPicker({
               <Menu.Item
                 icon={<Icon name="arrow_left_to_line" />}
                 onClick={handleStartingFromClick}
-                onMouseDown={handleStartingFromClick}
               >
                 {t`Starting from…`}
               </Menu.Item>
@@ -114,7 +110,6 @@ export function DateIntervalPicker({
             <Menu.Item
               icon={<Icon name={includeCurrent ? "check" : "calendar"} />}
               onClick={handleIncludeCurrentClick}
-              onMouseDown={handleIncludeCurrentClick} // we need to close the menu on mousedown to prevent the outer popover from closing
               aria-selected={includeCurrent}
               data-testid="include-current-interval-option"
             >
```

Reply:
Related: https://github.com/mantinedev/mantine/issues/4952#issuecomment-1743653834

Reply:
Great fix, applied